core/src/main/java/jenkins/model/MasterBuildConfiguration.java

```diff
@@ -51,12 +51,15 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         Jenkins j = Jenkins.get();
         try {
             // for compatibility reasons, this value is stored in Jenkins
-            String num = json.getString("numExecutors");
-            if (!num.matches("\\d+")) {
-                throw new FormException(Messages.Hudson_Computer_IncorrectNumberOfExecutors(),"numExecutors");
+            if(j.hasPermission(Jenkins.ADMINISTER)){
+                String num = json.getString("numExecutors");
+                if (!num.matches("\\d+")) {
```

Comment:
Nit: I know you're just refactoring here, but this could be done a bit more easily via `Integer.parseInt()` and catching the exception.